### PR TITLE
Prepare for 1.2.0-rc1 release

### PR DESCRIPTION
**Pull request by varunpuranik**
# 1.2.0-rc1 (2020-11-09)
* Preview support for nesting IoT Edge devices in gateway configuration, to allow creation of hierarchies of IoT Edge devices.
* Preview support for MQTT 3.1.1 compliant broker in EdgeHub. 
* Updates to `iotedge check` troubleshooting command to work in hierarchical configuration.

### Features
* Connect to parent IoT Edge device in a hierarchical configuration. ([b92785c](https://github.com/Azure/iotedge/commit/b92785c2fa1b123daf3f9a21b5f7c2f4110f9b19))
* Upload module logs and support bundle in a hierarchical configuration. ([37e8d08](https://github.com/Azure/iotedge/commit/37e8d08ba0af6571f629b7606c518dcc24e81ca6))
* Pull docker container images in a hierarchical configuration ([e82200d](https://github.com/Azure/iotedge/commit/e82200d31ad5745e7a8cb75abd99005ff314bede))

### Features
* Authenticate clients in a hierarchical configuration, including child IoT Edge devices that may be connecting on behalf of their children ([32a4e06](https://github.com/Azure/iotedge/commit/32a4e06ee0e4538210caa551169970c17f61bde0))
* Added module creation APIs to allow child IoT Edge devices to create modules in IoT Hub ([6336d92](https://github.com/Azure/iotedge/commit/6336d9209a56716fdcda29722e4c2bed451029a8))
* MQTT 3.1.1 compliant broker ([eb4a8cb](https://github.com/Azure/iotedge/commit/eb4a8cb9554ac48bc5c7954853cf4b211e4e37a2))
* Support for setting authorization policies for custom MQTT topics ([5effde9](https://github.com/Azure/iotedge/commit/5effde9b6d261cd1368943138191850a2ff0d465))
* Support for bridging MQTT topics to MQTT Broker in parent IoT Edge device ([5a79646](https://github.com/Azure/iotedge/commit/5a796463c97c3831990ed5aba58d06ee79049ca7))

### Features
* Configuration updates to support nesting IoT Edge devices ([b92785c](https://github.com/Azure/iotedge/commit/b92785c2fa1b123daf3f9a21b5f7c2f4110f9b19))
* Updates to `iotedge check` troubleshooting command to work in hierarchical configuration ([c0bad52](https://github.com/Azure/iotedge/commit/c0bad527da979fc0d8d1c810474e5078dfee83ca), [24b1c78](https://github.com/Azure/iotedge/commit/24b1c78f835068de7795f960660d45a889b4ae1b))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,27 @@
+# 1.2.0-rc1 (2020-11-06)
+* Preview support for daisy chaining IoT Edge devices in gateway configuration, to allow creation of hierarchies of IoT Edge devices.
```
Comment (ebertrams) [inline CHANGELOG.md:2]: **nesting** instead of **daisy chaining**? this is what we have used in the docs

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,27 @@
...
+### Features
+* Authenticate clients in a hierarchical configuration, including child IoT Edge devices that may be connecting on behalf of their children ([32a4e06](https://github.com/Azure/iotedge/commit/32a4e06ee0e4538210caa551169970c17f61bde0))
+* Added module creation APIs to allow child IoT Edge devices to create modules in IoT Hub ([6336d92](https://github.com/Azure/iotedge/commit/6336d9209a56716fdcda29722e4c2bed451029a8))
+* MQTT 3.1.1 compliant broker ([eb4a8cb](https://github.com/Azure/iotedge/commit/eb4a8cb9554ac48bc5c7954853cf4b211e4e37a2))
+* Support for setting authorization policies for custom MQTT topics ([5effde9](https://github.com/Azure/iotedge/commit/5effde9b6d261cd1368943138191850a2ff0d465))
+* Support for bridging MQTT topics to MQTT Broker in parent IoT Edge device ([5a79646](https://github.com/Azure/iotedge/commit/5a796463c97c3831990ed5aba58d06ee79049ca7))
+
+## iotedged
+### Features
+* Configuration updates to support daisy chaining ([b92785c](https://github.com/Azure/iotedge/commit/b92785c2fa1b123daf3f9a21b5f7c2f4110f9b19))
```
Comment (ebertrams) [inline CHANGELOG.md:22]: **nesting** instead of **daisy chaining**?

File: edgelet/version.txt [hunk 3]
```diff
@@ -1 +1 @@
-1.2.0~dev
+1.2.0-rc1
```
Comment (yophilav) [inline edgelet/version.txt:1]: `1.2.0~rc1`

Note the `~` not the `-`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
+# 1.2.0-rc1 (2020-11-06)
```
Comment (yophilav) [inline CHANGELOG.md:1]: Since we are releasing this Monday, we might as well use the date here : 2020-11-09

**Comment by yophilav** [comment]
Please update the PR description with the content of the changelog.

**Review by yophilav — APPROVED** [review]

**Pull request merged**